File: week-6/py/member_system/repository/memory_repository/memory_unit_of_work.py
from member_system.repository.unit_of_work import MemberRepository, MessageRepository, UnitOfWork
from member_system.repository.memory_repository.memory_member_repository import MemoryMemberRepository
from member_system.repository.memory_repository.memory_message_repository import MemoryMessageRepository

class MemoryUnitOfWork(UnitOfWork):
    def _createMemberRepository(self) -> MemberRepository:
        return MemoryMemberRepository()
    
    def _createMessageRepository(self) -> MessageRepository:
        return MemoryMessageRepository(self.memberRepository)

File: week-6/go/membersys/repository/memory_unit_of_work.go
package repository

type MemoryUnitOfWork struct {
	UnitOfWork
}

func (u *MemoryUnitOfWork) init() {
	memberRepository := MemoryMemberRepository{}
	u.MemberRepository = &memberRepository
	u.MessageRepository = &MemoryMessageRepository{memberRepository: &memberRepository}
}

func NewMemoryUnitOfWork() *MemoryUnitOfWork {
	unitOfWork := MemoryUnitOfWork{}
	InitUnitOfWork(&unitOfWork)
	return &unitOfWork
}
